pkg/services/store/entity/sqlstash/broadcaster.go

```diff
@@ -23,101 +102,132 @@ func NewBroadcaster[T any](ctx context.Context, connect ConnectFunc[T]) (Broadca
 }
 
 type broadcaster[T any] struct {
-	running     bool // FIXME: race condition between `Subscribe`/`Unsubscribe` and `start`
-	ctx         context.Context
-	subs        map[chan T]struct{}
```

Comment:
`subs` changed from `map[chan T]struct{}` to `map[<-chan T]chan T` to allow indexing by subscriptor while still being able to send and close the channel. This could be replaced by an opaque cookie to prevent access to the reading side and hide the implementation details, but it's not necessary now.